routers/api/v1/repo/pull.go

```diff
@@ -508,6 +510,30 @@ func EditPullRequest(ctx *context.APIContext, form api.EditPullRequestOption) {
 		notification.NotifyIssueChangeStatus(ctx.User, issue, statusChangeComment, issue.IsClosed)
 	}
 
+	// change pull target branch
+	if len(form.Base) != 0 && form.Base != pr.BaseBranch {
+		if !ctx.Repo.GitRepo.IsBranchExist(form.Base) {
+			ctx.Error(http.StatusNotFound, "NewBaseBranchNotExist", fmt.Errorf("new base '%s' not exist", form.Base))
+			return
+		}
+		if err := pull_service.ChangeTargetBranch(pr, ctx.User, form.Base); err != nil {
+			if models.IsErrPullRequestAlreadyExists(err) {
+				ctx.Error(http.StatusConflict, "IsErrPullRequestAlreadyExists", err)
+				return
+			} else if models.IsErrIssueIsClosed(err) {
+				ctx.Error(http.StatusUnprocessableEntity, "IsErrPullRequestAlreadyExists", err)
+				return
+			} else if models.IsErrPullRequestHasMerged(err) {
+				ctx.Error(http.StatusConflict, "IsErrPullRequestAlreadyExists", err)
+				return
+			} else {
+				ctx.InternalServerError(err)
```

Comment:
```suggestion
			} else if models.IsErrBranchesEqual(err) {
				ctx.Error(http.StatusUnprocessableEntity, "IsErrBranchesEqual", err)
			} else {
				ctx.InternalServerError(err)
```

Reply:
Not sure about `http.StatusUnprocessableEntity` vs `http.StatusConflict`

Reply:
Maybe instead of `ctx.InternalServerError`: `ctx.ServerError("UpdatePullRequestTarget", err)`?

Reply:
@blaggacao IsErrBranchesEqual should not happen since it is checked before:
> && form.Base != pr.BaseBranch

Reply:
ah, yep sorry